test/scripts/bridge/set-env.sh

```diff
@@ -0,0 +1,242 @@
+  local expected_version=$1
+
+  if ! [ -x "$(command -v node)" ]; then
+      echo 'Error: NodeJS is not installed.'
+      exit 1
+  fi
+
+  node_version=$(node -v) # This does not seem to work in Git Bash on Windows.
+  # "node -v" outputs version in the format "v18.12.1"
+  node_version=${node_version:1} # Remove 'v' at the beginning
+  node_version=${node_version%\.*} # Remove trailing ".*".
+  node_version=${node_version%\.*} # Remove trailing ".*".
+  node_version=$(($node_version)) # Convert the NodeJS version number from a string to an integer.
+  if [ $node_version -lt "$expected_version" ]
+  then
+    echo "NodeJS version is lower than $expected_version (it is $node_version), Please update your node installation!"
+    exit 1
+  fi
+}
+
+vercomp() {
+    if [[ $1 == $2 ]]
+    then
+        echo "Equal"
+        return
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if ((10#${ver1[i]:=0} > 10#${ver2[i]:=0}))
+        then
+            echo "Greater"
+            return
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            echo "Less"
+            return
+        fi
+    done
+}
+
+
+check_go_version() {
+  local expected_version=$1
+
+  if ! [ -x "$(command -v go)" ]; then
+        echo 'Error: Go is not installed.'
+        exit 1
+  fi
+
+  go_version=$(go version | { read _ _ v _; echo ${v#go}; })
+  op=$(vercomp "$go_version" "$1")
+
+  if [[ $op = "Less" ]]
+  then
+      echo "Go version is lower than $expected_version (it is $go_version), Please update your go installation!"
+      exit 1
+  fi
+}
+
+check_tool() {
+    if ! [ -x "$(command -v g++)" ]; then
+        echo 'Error: g++ is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v protoc)" ]; then
+        echo 'Error: protoc is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v jq)" ]; then
+        echo 'Error: jq is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v mage)" ]; then
+        echo 'Error: mage is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v pnpm)" ]; then
+        echo 'Error: pnpm is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v forge)" ]; then
+        echo 'Error: foundry is not installed.'
+        exit 1
+    fi
+    if ! [ -x "$(command -v yarn)" ]; then
+        echo 'Error: yarn is not installed.'
+        exit 1
+    fi
+    if [[ "$(uname)" == "Darwin" && -z "${IN_NIX_SHELL:-}" ]]; then
+          if ! [ -x "$(command -v gdate)" ]; then
+              echo 'Error: gdate (GNU Date) is not installed.'
+              exit 1
+          fi
+      else
+          if ! [ -x "$(command -v date)" ]; then
+              echo 'Error: date is not installed.'
+              exit 1
+          fi
+    fi
+
+    check_node_version 22
```

Comment:
I have node 20 so I changed this line and it works, where does the requirement to use node 22 come from?

Reply:
I just put the latest LTS version.